assets/js/stripe-payment-request.js

```diff
@@ -490,6 +424,159 @@ jQuery( function( $ ) {
+						window.alert( wc_add_to_cart_variation_params.i18n_unavailable_text );
+					} else if ( addToCartButton.is( '.wc-variation-selection-needed' ) ) {
+						window.alert( wc_add_to_cart_variation_params.i18n_make_a_selection_text );
+					}
+					return;
+				}
+
+				if ( 0 < paymentRequestError.length ) {
+					evt.preventDefault();
+					window.alert( paymentRequestError );
+					return;
+				}
+
+				wc_stripe_payment_request.addToCart();
+
+				if ( wc_stripe_payment_request.isCustomPaymentRequestButton( prButton ) ) {
+					evt.preventDefault();
+					paymentRequest.show();
+				}
+			});
+
+			$( document.body ).on( 'woocommerce_variation_has_changed', function () {
+				wc_stripe_payment_request.blockPaymentRequestButton( prButton );
+
+				$.when( wc_stripe_payment_request.getSelectedProductData() ).then( function ( response ) {
+					$.when(
+						paymentRequest.update( {
+							total: response.total,
+							displayItems: response.displayItems,
+						} )
+					).then( function () {
+						wc_stripe_payment_request.unblockPaymentRequestButton( prButton );
+					} );
+				});
+			} );
+
+			$( '.quantity' ).on( 'keyup', '.qty', function () {
+				wc_stripe_payment_request.blockPaymentRequestButton( prButton );
+				paymentRequestError = [];
+
+				$.when( wc_stripe_payment_request.getSelectedProductData() ).then( function ( response ) {
+					if ( response.error ) {
+						paymentRequestError = [ response.error ];
+						wc_stripe_payment_request.unblockPaymentRequestButton( prButton );
+					} else {
+						$.when(
+							paymentRequest.update( {
+								total: response.total,
+								displayItems: response.displayItems,
+							} )
+						).then( function () {
+							wc_stripe_payment_request.unblockPaymentRequestButton( prButton );
+						});
+					}
+				} );
+			});
+		},
+
+		attachCartPageEventListeners: function ( prButton, paymentRequest ) {
+			if ( ! wc_stripe_payment_request_params.button.is_custom || ! wc_stripe_payment_request.isCustomPaymentRequestButton( prButton ) ) {
+				return;
+			}
+
+			prButton.on( 'click', function ( evt ) {
+				evt.preventDefault();
+				paymentRequest.show();
+			} );
+		},
+
+		showPaymentRequestButton: function( prButton ) {
+			if ( wc_stripe_payment_request.isCustomPaymentRequestButton( prButton ) ) {
+				prButton.addClass( 'is-acitve' );
+				$( '#wc-stripe-payment-request-wrapper, #wc-stripe-payment-request-button-separator' ).show();
+			} else if ( $( '#wc-stripe-payment-request-button' ).length ) {
+				$( '#wc-stripe-payment-request-wrapper, #wc-stripe-payment-request-button-separator' ).show();
+				prButton.mount( '#wc-stripe-payment-request-button' );
+			}
+		},
+
+		blockPaymentRequestButton: function( prButton ) {
+			$( '#wc-stripe-payment-request-button' ).block( { message: null } );
+			if ( wc_stripe_payment_request.isCustomPaymentRequestButton( prButton ) ) {
+				prButton.addClass( 'is-blocked' );
```

Comment:
Here we are allowing theme developers to customize the PR button **only** if the button is set to "custom". Why not always add the `is-blocked` class when the button is blocked?

I'm working on https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1539 and I was wondering if there is a reason why we do that only for custom PR buttons. Check my [comment](https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1539#issuecomment-827909578) for more context

Reply:
I think it wasn't necessary for our own buttons :) I also think it won't harm if the class is applied every time. But you'd need to make user the class is removed every time when the button is unblocked.

Reply:
@vbelolapotkov thanks for response.

>  it won't harm if the class is applied every time. But you'd need to make sure the class is removed every time when the button is unblocked.

 Yes. That's how it was done.